spec.md

```diff
@@ -2334,6 +2334,46 @@ The width of each primitive operation is detailed in [@sec:primitive-operations]
 The width of the integer literal expressions is detailed in their respective
 sections.
 
+# Combinational Loops
+
+Combinational logic is a section of logic with no registers between gates.
+A combinational loop exists when the output of some combinational logic 
+is fed back into the input of that combinational logic with no intervening
+register. FIRRTL does not support combinational loops even if it is possible
+to show that the loop does not exist under actual mux select values.
+Combinational loops are not allowed and it shouldn't depend on any FIRRTL
+transformation to remove or break such combinational loops.
+
+The following example is not legal, since a combinational loop exists, 
+`w -> b -> w`, even it can be proved that the loop does not exist after
+some simplification.
+
+``` firrtl
+wire a: UInt
+wire b: UInt
+wire c: UInt<1>
+wire notC: UInt<1>
+wire w: UInt
+w <= a
+when c :
+  w <= b
+notC <= not(c)
+when notC :
+  b <= w
+```
```

Comment:
I think this can be more plainly stated that combinational loops that would be removed by last connect semantics, conditional last connect semantics, or optimizations are still flagged as loops. The most basic example which shows a loop that would be removed by last connect semantics is:

```
circuit Foo:
  module Foo:
    input a: UInt<1>
    output b: UInt<1>
    b <= b
    b <= a
```

Reply:
Thanks for the suggestion. Updated the example.